v2/pkg/contextargs/contextargs.go

```diff
@@ -0,0 +1,19 @@
+package contextargs
+
+import (
+	"net/http/cookiejar"
+	"sync"
+)
+
+type Context struct {
+	Input     string
+	CookieJar *cookiejar.Jar
+
+	// Access to Args must use lock strategies to prevent data races
+	*sync.RWMutex
+	Args map[string]interface{}
+}
+
+func New() Context {
```

Comment:
Maybe by default we should accept a input.

Reply:
done